torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1446,6 +1510,7 @@ c10::intrusive_ptr<ProcessGroup::Work> ProcessGroupNCCL::collective(
     Fn fn,
     OpType opType,
     const char* profilingTitle) {
+  markRankStuck(rankStatusHelper_.get(), rank_ % at::cuda::getNumGPUs());
```

Comment:
nit: I don't think this should be called `markRankStuck`, but probably something like `markRankProcessing` or something.

Reply:
Also, I think we should do the same for `pointToPoint` as well since ranks can potentially get stuck there too.